tests/net/socket/tls/src/main.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2020 Friedt Professional Engineering Services, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+#include <net/net_core.h>
+#include <net/net_ip.h>
+#include <net/socket.h>
+#include <net/tls_credentials.h>
+#include <posix/unistd.h>
+#include <ztest.h>
+
+LOG_MODULE_REGISTER(tls_test, CONFIG_NET_SOCKETS_LOG_LEVEL);
+
+#define STACK_SIZE 1024
+
+#define PORT 4242
+
+/* The answer to life, the universe, and everything */
+#define SECRET "forty-two"
+#define SECRET_SIZE 9
```

Comment:
Better would be `#define SECRET_SIZE (sizeof(SECRET) - 1)` so that if the SECRET is changed, the length is automatically calculated correctly.